Lib/annotationlib.py

```diff
@@ -267,18 +281,25 @@ def __init__(self, node, globals=None, owner=None, is_class=False, cell=None):
         self.__cell__ = cell
         self.__owner__ = owner
 
-    def __convert(self, other):
+    def __convert_to_ast(self, other):
         if isinstance(other, _Stringifier):
+            if isinstance(other.__ast_node__, str):
+                return ast.Name(id=other.__ast_node__)
             return other.__ast_node__
         elif isinstance(other, slice):
             return ast.Slice(
-                lower=self.__convert(other.start) if other.start is not None else None,
-                upper=self.__convert(other.stop) if other.stop is not None else None,
-                step=self.__convert(other.step) if other.step is not None else None,
+                lower=self.__convert_to_ast(other.start) if other.start is not None else None,
+                upper=self.__convert_to_ast(other.stop) if other.stop is not None else None,
+                step=self.__convert_to_ast(other.step) if other.step is not None else None,
             )
         else:
             return ast.Constant(value=other)
 
+    def __get_ast(self):
+        if isinstance(self.__ast_node__, str):
+            return ast.Name(id=self.__ast_node__)
+        return self.__ast_node__
```

Comment:
You might add a variable to not have to get/reference the attribute 3 times:

```suggestion
        node = self.__ast_node__
        if isinstance(node, str):
            return ast.Name(id=node)
        return node
```